test/parallel/test-util-text-encoder.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { TextEncoder, TextDecoder } = require('util');
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+assert.strictEqual(decoder.decode(encoder.encode('')), '');
+assert.strictEqual(decoder.decode(encoder.encode('latin1')), 'latin1');
+assert.strictEqual(decoder.decode(encoder.encode('Yağız')), 'Yağız');
+
+const dest = new Uint8Array(5).fill(0);
+const encoded = encoder.encodeInto('latin', dest);
+assert.strictEqual(encoded.read, dest.length);
+assert.strictEqual(encoded.written, dest.length);
+assert.strictEqual(decoder.decode(dest), 'latin');
```

Comment:
A test that verifies what happens if the `dest` buffer is too small and too big for the UTF8 encoded result would be good.

Reply:
Thanks for the review @jasnell. I just updated the pull request